core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -577,7 +577,7 @@ public void start(Boolean isPrimary) throws IOException {
           mPersistRequests.put(inodeFile.getId(), new alluxio.time.ExponentialTimer(
               ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_INITIAL_INTERVAL_MS),
               ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_INTERVAL_MS),
-              ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_INITIAL_WAIT_TIME_MS),
+              inodeFile.getPersistenceWaitTime(),
```

Comment:
This calculation needs to be a bit different on recovery since the file should not be treated as created on restart (for example if wait time is 1 day and restart happened 23hrs after file creation, we should not wait 1 more day)

Reply:
Maybe instead of the persist wait time, I use a `should persist time(create file time + persist wait time)`. In here I can pass in the`(should persist time - current time)`. If it's smaller than 0, than I set 0 wait time.

Reply:
That sounds reasonable.